kernel/work.c

```diff
@@ -736,6 +791,14 @@ void k_work_queue_run(struct k_work_q *queue, const struct k_work_queue_config *
 		k_thread_name_set(_current, cfg->name);
 	}
 
+#if defined(CONFIG_WORKQUEUE_WORK_TIMEOUT)
+	if ((cfg != NULL) && (cfg->work_timeout_ms)) {
+		queue->work_timeout = K_MSEC(cfg->work_timeout_ms);
```

Comment:
In case k_timeout_t is preferred

```suggestion
	if (cfg != NULL) {
		queue->work_timeout = cfg->work_timeout;
```